include/ada/url-inl.h

```diff
@@ -97,25 +97,27 @@ namespace ada {
         }
       }
 
-      out.host_end = out.host_start + host.value().size() - 1;
+      out.host_end = uint32_t(out.host_start + host.value().size() - 1);
     }
 
-    out.port = port;
     out.pathname_start = out.host_end;
 
     if (port.has_value()) {
+      out.port = out.host_end;
       out.pathname_start += std::to_string(port.value()).size();
+    } else {
+      out.port = ada::url_components::omitted;
```

Comment:
Port has a default value of `omitted` right? We don't need this line.

Reply:
Corect.